code/datums/helper_datums/teleport.dm

```diff
@@ -219,6 +219,19 @@
 			teleatom.visible_message("<span class='danger'>\The [teleatom] bounces off of the portal!</span>")
 		return FALSE
 
+	// Block teleportation to/from/within z-level 7 (procedurally-generated planets)
+	var/turf/curturf = get_turf(teleatom)
+	if(curturf && curturf.z == map.zProcGen)
+		if(ismob(teleatom))
+			var/mob/M = teleatom
+			to_chat(M, "<span class='warning'>A mysterious force prevents teleportation from this location!</span>")
+		return FALSE
+	if(destination.z == 7)
```

Comment:
Should be a way to override that for buses and shit. Or alternatively, check the teleportation on a planet-by-planet level, with some planet allowed teleportation (if necessary, when something was done, like laying down a beacon or something) and some other notes.

Reply:
I plan to allow teleportation to these once I implement actual v-levels and make telescience compatible with the new/spoofed coords. I believe admins can still make a gateway on the planets, though I'd have to test that.